lib/std/os/windows.zig

```diff
@@ -1528,6 +1528,81 @@ pub const PathSpace = struct {
     }
 };
 
+/// The error type for `removeDotDirsSanitized`
+pub const RemoveDotDirsError = error { TooManyParentDirs };
+
+/// Removes '.' and '..' path components from a "sanitized relative path".
```

Comment:
this function looks like it belongs in std.fs.path. it might be a duple of `resolveWindows`?

Reply:
It's specific to windows...only works with backslashes.  We could also make it work with forward slashes but I think posix has other better way to perform this operation.  This function doesn't touch the filesystem or use `CWD` to normalize the path like `resolveWindows` does.

Reply:
> It's specific to windows...only works with backslashes

Os-specific code is all over the standard library: it does all just get dumped into os/windows.zig.